src/plugins/list/list.c

```diff
@@ -0,0 +1,365 @@
+	key = ksLookupByName(config, "/placements/get", 0);
+	if(key)
+	{
+		const char *getString = keyString(key);
+		const char *getStrings[] = {"pregetstorage", "postgetstorage"};
+		GetPlacements getPlacement = preGetStorage;
+		while(getPlacement != getEnd)
+		{	
+			if(strstr(getString, getStrings[getPlacement]))
+			{
+				placements->getPlacements[getPlacement] = 1;
+			}
+			++getPlacement;
+		}
+	}
+	key = ksLookupByName(config, "/placements/err", 0);
+	if(key)
+	{
+		const char *errString = keyString(key);
+		const char *errStrings[] = {"prerollback", "postrollback"};
+		ErrPlacements errPlacement = preRollback;
+		while(errPlacement != errEnd)
+		{	
+			if(strstr(errString, errStrings[errPlacement]))
+			{
+				placements->errPlacements[errPlacement] = 1;
+			}
+			++errPlacement;
+		}
+	}
+	key = ksLookupByName(config, "/plugins", 0);
+	Key *cur; 
+	KeySet *cutKS = ksCut(config, key);
+	ksRewind(cutKS);
+	while((cur = ksNext(cutKS)) != NULL)
+	{
+		if(keyRel(key, cur) != 1)
+			continue;
+		Key *sub;
+		Key *lookup = keyDup(cur);
+		keyAddBaseName(lookup, "placements");
+		keyAddBaseName(lookup, "set");
+		sub = ksLookup(cutKS, lookup, 0);
+		if(sub)
+		{
+			const char *setString = keyString(sub);
+			const char *setStrings[] = {"presetstorage", "precommit", "postcommit"};
+			SetPlacements setPlacement = preSetStorage;
+			while(setPlacement != setEnd)
+			{	
+				if(strstr(setString, setStrings[setPlacement]))
+				{
+					ksAppendKey(placements->setKS[setPlacement], keyDup(cur));
+				}
+				++setPlacement;
+			}
+		}	
+		keySetBaseName(lookup, "get");
+		sub = ksLookup(cutKS, lookup, 0);
+		if(sub)
+		{
+			const char *getString = keyString(sub);
+			const char *getStrings[] = {"pregetstorage", "postgetstorage"};
+			GetPlacements getPlacement = preGetStorage;
+			while(getPlacement != getEnd)
+			{	
+				if(strstr(getString, getStrings[getPlacement]))
+				{
+					ksAppendKey(placements->getKS[getPlacement], keyDup(cur));
+				}
+				++getPlacement;
+			}
+		}	
+		keySetBaseName(lookup, "err");
+		sub = ksLookup(cutKS, lookup, 0);	
+		if(sub)
+		{
+
+			const char *errString = keyString(sub);
+			const char *errStrings[] = {"preRollback", "postRollback"};
+			ErrPlacements errPlacement = preRollback;
+			while(errPlacement != errEnd)
+			{	
+				if(strstr(errString, errStrings[errPlacement]))
+				{
+					ksAppendKey(placements->errKS[errPlacement], keyDup(cur));
+				}
+				++errPlacement;
+			}
+		}
+		keyDel(lookup);
+	}
+	ksDel(cutKS);
+	ksDel(config);
+	placements->initialized = 1;
+	return 1; /* success */
+}
+
+int elektraListClose(Plugin *handle , Key *errorKey ELEKTRA_UNUSED)
+{
+	/* free all plugin resources and shut it down */
+	Placements *placements = elektraPluginGetData(handle);
+	ksDel(placements->getKS[0]);
+	ksDel(placements->getKS[1]);
+	ksDel(placements->setKS[0]);
+	ksDel(placements->setKS[1]);
+	ksDel(placements->setKS[2]);
+	ksDel(placements->errKS[0]);
+	ksDel(placements->errKS[1]);
+	elektraFree(placements);
```

Comment:
elektraPluginGetData still points somewhere. So that elektraListOpen would work again, you would have to elektraPluginSetData again. (At least elektraListOpen looks like that this was intended)